torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -33,14 +33,26 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
   if (!variable.requires_grad())
     return {};
 
-  auto new_grad = grads[0];
+  auto& new_grad = grads[0];
```

Comment:
I don't think this is correct anymore. If there are variable hooks, this will get assigned a reference to the memory of a temporary vector that the hook returned, and will outlive it. That's undefined behavior. We should really just `std::move(grads[0])` here

Reply:
std::move should work but I think reference works too.since you cannot reassign reference in c++, the hook line is basically copying hook_returned_vector[0] into grads[0] and new_grad is still reference to grads[0].